doc/shared/suncontrol/SUNControl_Description.rst

```diff
@@ -0,0 +1,480 @@
+
+Each of the following routines are *optional* for any specific SUNControl
+implementation, however some may be required based on the implementation's
+SUNControl_ID (see table :numref:`SUNControl.Description.controllerIDs`). We
+note these requirements, as well as the behavior of the base SUNControl wrapper
+routine, below.
+
+.. c:function:: SUNControl_ID SUNControlGetID(SUNControl C)
+
+   Returns the type identifier for the controller *C*. Returned values
+   are given in :numref:`SUNControl.Description.controllerIDs`
+
+   :param C: the :c:type:`SUNControl` object.
+   :return: :c:type:`SUNControl_ID` type identifier.
+
+   Usage:
+
+   .. code-block:: c
+
+      SUNControl_ID id = SUNControlGetID(C);
+
+.. c:function:: void SUNControlDestroy(SUNControl C)
+
+   Deallocates the controller *C*. If this is not provided by the
+   implementation, the base wrapper routine will free both the *content* and
+   *ops* objects -- this should be sufficient unless a controller implementation
+   performs dynamic memory allocation of its own (note that the
+   SUNDIALS-provided SUNControll implementations do not need to supply this
+   routine).
+
+   :param C: the :c:type:`SUNControl` object.
+
+   Usage:
+
+   .. code-block:: c
+
+      SUNControlDestroy(C);
+
+.. c:function:: int SUNControlEstimateStep(SUNControl C, realtype h, realtype dsm, realtype* hnew)
+
+   Estimates a single-rate step size. This routine is required for controllers
+   of type ``SUNDIALS_CONTROL_H``.
+
+   :param C: the :c:type:`SUNControl` object.
+   :param h: the step size from the previous step attempt.
+   :param dsm: the local temporal estimate from the previous step attempt.
+   :param hnew: (output) pointer to the estimated step size.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlEstimateStep(C, hcur, dsm, &hnew);
+
+.. c:function:: int SUNControlEstimateStepAndOrder(SUNControl C, realtype h, int q, realtype dsm, realtype* hnew, int* qnew)
+
+   Estimates a single-rate step size and corresponding method order. This
+   routine is required for controllers of type ``SUNDIALS_CONTROL_HQ``.
+
+   :param C: the :c:type:`SUNControl` object..
+   :param h: the step size from the previous step attempt.
+   :param q: the method order from the previous step attempt.
+   :param dsm: the local temporal estimate from the previous step attempt.
+   :param hnew: (output)  pointer to the estimated step size.
+   :param qnew: (output)  pointer to the estimated method order.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlEstimateStepAndOrder(C, hcur, qcur, dsm, &hnew, &qnew);
+
+.. c:function:: int SUNControlEstimateMRISteps(SUNControl C, realtype H, realtype h, realtype DSM, realtype dsm, realtype* Hnew, realtype *hnew)
```

Comment:
Potentially, however I find it unlikely that with highly nested MRI calculations, anyone would want "adaptivity" that forces fixed values for `h[0]`, `h[1]`, ... `h[maxl]` throughout the entire "slow" MRI step.

Instead, I envision that for nested MRI it should be preferable to use the H-Tol controllers, since these may naturally be nested without any changes, and would allow varying the "fastest" step size throughout a given "slowest" step.  

If you would like to investigate MRI adaptivity with multiple fixed step sizes values throughout all nested levels within a given slow step, then I can update this to be more flexible.  However, I do not intend to pursue that direction.